ci/owncloud/upload_to_owncloud.py

```diff
@@ -0,0 +1,153 @@
+    """
+    upload(webdav_client, is_direct, remote_path, local_path, unique_id)
+
+    Uploads the content of a directory to a directory in the remote webdav server
+
+    The remote directory's tail is a unique directory name.
+
+    Parameters
+    ----------
+
+    webdav_client : webdav3.client.Client
+        the remote dav server's client object
+    is_direct : bool
+        flag dictating whether the directory is uploaded directly into the remote path
+        or copied into a temporary folder first, and then into the remote path.
+
+    remote_path : str
+        path in the webdav server to be created and for the files to be copied to
+
+    local_path : str
+        local path to be copied from
+
+    unique_id : str, optional
+        a uuid to act as the identifier of the uploaded files in the remote directory
+    """
+    if not unique_id:
+        unique_id = str(uuid4())
+    message("""Uploading files from local directory {0} to remote directory {1},\n \
+         username is {2}, remote directory id is: {3}""".format(
+        local_path, remote_path, webdav_client.webdav.options["login"], webdav_client, unique_id))
+    remote_full_path = path.join(remote_path, unique_id)
+    mkdir_recursively(remote_full_path)
+    if not path.isdir(local_path):
+        raise FileNotFoundError("Local path location is not a directory")
+    if is_direct:
+        webdav_client.upload_sync(remote_full_path, local_path)
+    else:
+        remote_temp_path = path.join("temp_uploads", unique_id)
+        mkdir_recursively(remote_temp_path)
+        webdav_client.upload_sync(remote_temp_path, local_path)
+        webdav_client.copy(remote_temp_path, remote_full_path)
+    message("Success")
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--verbose", "-v", action='store_true', default=False,
+                        help="Enable verbosity.")
+    parser.add_argument("--direct", "-d", action='store_true', default=False,
+                        help="Direct upload without first uploading to a temporary directory")
+    parser.add_argument("--url", "-u", action='store', nargs=1,
+                        default="https://ftp.essensium.com/owncloud/remote.php/dav/files/{0}"
+                        .format(getuser()),
+                        help="webDAV URL")
+    parser.add_argument("local-path", nargs=1, type=str, action='store',
+                        help="Local file or folder to upload")
+    parser.add_argument("remote-path", type=str, nargs=1, action='store',
+                        help="Path in the cloud, relative to the user home to upload")
+    args = parser.parse_args()
+    args.local_path = getattr(args, "local-path")[0]
+    args.remote_path = getattr(args, "remote-path")[0]
+    opts.verbose = args.verbose
+    webdav_options = get_credentials()
+    client = Client(webdav_options)
```

Comment:
(nitpick) I would move the creation of the client inside the upload function, i.e. pass the dictionary as the first argument. That makes it easier to use this function by importing.